src/cargo/ops/cargo_install.rs

```diff
@@ -385,7 +386,7 @@ impl<'gctx> InstallablePackage<'gctx> {
                 .iter()
                 .filter(|t| t.is_executable())
                 .collect();
-            if !binaries.is_empty() {
+            if !binaries.is_empty() && !dry_run {
```

Comment:
We should not bypass this warning on dry-run as this could affect people's decisions for what command-line arguments they want to pass

Reply:
Fixed in 86415010617b63c981fef8c18328331e11d47434